### PR TITLE
TileTypes now use the PrototypeManager and fixes

### PR DESCRIPTION
**Pull request by TomMalbran**
1. TileTypes now use the PrototypeManager.
2. Fixed a couple of issues with placing Ladders.

### CODE REVIEW

File: Assets/Scripts/UI/DialogBox/Settings/DialogBoxSettings.cs [hunk 1]
```diff
@@ -78,7 +78,7 @@ public void OnClickSave()
         this.CloseDialog();
         WorldController.Instance.spawnInventoryController.SetUIVisibility(developerModeToggle.isOn);
         LocalizationTable.SetLocalization(languageDropdown.value);
-        //FurnitureBuildMenu.instance.RebuildMenuButtons(developerModeToggle.isOn);
+        ///FurnitureBuildMenu.instance.RebuildMenuButtons(developerModeToggle.isOn);
```
Comment (svmnotn) [inline Assets/Scripts/UI/DialogBox/Settings/DialogBoxSettings.cs]: this should probably just be a blank space, or deleted since git tracks lines like this

Comment (TomMalbran) [inline Assets/Scripts/UI/DialogBox/Settings/DialogBoxSettings.cs]: This works for StyleCop (was only fixing a warning), I don't think that it needs changed. At some point we will need to fix that error and call a similar function.

Comment (sboigelot) [inline Assets/Scripts/UI/DialogBox/Settings/DialogBoxSettings.cs]: Yes you should replace your extra / by a extra space. Comment should start by a space, don't ask me why, stylecop rule.

Comment (TomMalbran) [inline Assets/Scripts/UI/DialogBox/Settings/DialogBoxSettings.cs]: StyleCop is fine with the `///` comments. It has issues with `//`. I changed it anyway and added a TODO

**Comment by sboigelot** [comment]
@TomMalbran I'm getting a very weird error on your PrototypeManager with stylecop. This is the same on master but I was wondering if you had the issue:
```

Warning	2	SA0001 : CoreParser : An exception occurred while parsing the file: System.FormatException, Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(IFormatProvider provider, String format, Object[] args)
   at StyleCop.StyleCopThread.RunAnalyzers(CodeDocument document, SourceParser parser, IEnumerable`1 analyzers)
   at StyleCop.StyleCopThread.TestAndRunAnalyzers(CodeDocument document, SourceParser parser, IEnumerable`1 analyzers, Int32 passNumber)
   at StyleCop.StyleCopThread.ParseAndAnalyzeDocument(SourceCode sourceCode, DocumentAnalysisStatus documentStatus)
   at StyleCop.StyleCopThread.DoWork(Object sender).	D:\git\ProjectPorcupine\Assets\Scripts\Models\Prototypes\PrototypeMap.cs	1	1	ProjectPorcupine.CSharp

```

There seems to be an old stylecop bug about object array but I don't manage to pinpoint the issue.

**Comment by TomMalbran** [comment]
I dont get any error from StyleCop. I use MonoDevelop that comes with Unity, since I use a Mac.

**Comment by sboigelot** [comment]
I'll investigate later, if any other tester on windows can have a look too it would be great!

**Comment by koosemose** [comment]
No stylecop bugs showing up on Win7 + MonoDevelop for me.

File: Assets/Scripts/Models/TileType.cs [hunk 2]
```diff
@@ -161,36 +145,35 @@ public override string ToString()
         return Type;
     }
 
-    // Checks whether the given floor type is allowed to be built on the tile.
-    public bool CanBuildHere(Tile t)
+    /// <summary>
+    /// Determines whether this tile type is allowed to be built on the given tile.
+    /// </summary>
+    /// <returns><c>true</c> if the ile type is allowed to be built on the given tile; otherwise, <c>false</c>.</returns>
+    /// <param name="t">The tile to build on.</param>
```
Comment (Grenkin1988) [inline Assets/Scripts/Models/TileType.cs]: `/// <param name="tile">The tile to build on.</param>`

File: Assets/Scripts/Models/TileType.cs [hunk 3]
```diff
@@ -161,36 +145,35 @@ public override string ToString()
...
-    public System.Xml.Schema.XmlSchema GetSchema()
-    {
-        return null;
-    }
-
-    public void ReadXml(XmlReader parentReader)
+    /// <summary>
+    /// Reads the prototype from the specified XML reader.
+    /// </summary>
+    /// <param name="readerParent">The XML reader to read from.</param>
```
Comment (Grenkin1988) [inline Assets/Scripts/Models/TileType.cs]: `parentReader`

**Comment by TomMalbran** [comment]
Fixed the comments.

**Comment by koosemose** [comment]
Conflicted

**Comment by TomMalbran** [comment]
Conflicts fixed.

**Pull request merged**